src/hatch/template/files_default.py

```diff
@@ -136,23 +136,86 @@ def __init__(self, template_config: dict, plugin_config: dict):
 
 [tool.hatch.envs.default]
 dependencies = [
+  "coverage[toml]>=6.5",
   "pytest",
-  "pytest-cov",
 ]
 [tool.hatch.envs.default.scripts]
-cov = "pytest --cov-report=term-missing --cov-config=pyproject.toml --cov={package_location}{template_config['package_name']} --cov=tests {{args}}"
-no-cov = "cov --no-cov {{args}}"
+test = "pytest {{args:tests}}"
+test-cov = "coverage run -m pytest {{args:tests}}"
+cov-report = [
+  "- coverage combine",
+  "coverage report",
+]
+cov = [
+  "test-cov",
+  "cov-report",
+]
 
-[[tool.hatch.envs.test.matrix]]
+[[tool.hatch.envs.all.matrix]]
 python = ["3.7", "3.8", "3.9", "3.10", "3.11"]
 
+[envs.lint]
+detached = true
+dependencies = [
+  "black",
+  "mypy",
+  "ruff",
+]
+[envs.lint.scripts]
+typing = "mypy --install-types --non-interactive {{args:{package_location}{template_config['package_name']} tests}}"
+style = [
+  "ruff {{args:.}}",
+  "black --check --diff {{args:.}}",
+]
+fmt = [
+  "black {{args:.}}",
+  "ruff --fix {{args:.}}",
+  "style",
+]
+all = [
+  "style",
+  "typing",
+]
+
+[tool.black]
+target-version = ["py37"]
+line-length = 120
+skip-string-normalization = true
+
+[tool.ruff]
+target-version = "py37"
+line-length = 120
+select = ["A", "B", "C", "E", "F", "FBT", "I", "M", "N", "Q", "RUF", "S", "T", "U", "W", "YTT"]
```

Comment:
Looks great to me! Ruff's internal default is `["E", "F"]` so this is expanded to include most rules, which is of course fine with me but totally up to Hatch :)

Reply:
Is there a way to select all by default?

Reply:
Not right now, no, but we're looking into adding "presets" which would solve this problem (https://github.com/charliermarsh/ruff/issues/809).

(Selecting all could also cause issues because, e.g., some of the docstring rules (`D`, omitted here) are incompatible with one another, like "Must have exactly one blank line after summary" vs. "Must have exactly zero blank lines after summary".)

Reply:
Also gonna add:

```toml
# Tests can use assertions
"tests/*" = ["S101"]
"tests/**/*" = ["S101"]
```